### PR TITLE
doc: add indexes with weight description

### PR DESCRIPTION
**Pull request by kenhys**

### CODE REVIEW

File: doc/source/tutorial/match_columns.rst [hunk 1]
```diff
@@ -290,6 +290,18 @@ On the other hand, the second one matches one article only because of ``Replies2
 Indexes with Weight
 -------------------
 
-TODO
+You can search blog entries which contains important keyword in ``Blog1`` table.
+For example, let's try to search blog entries which contains ``Groonga`` as important keyword.
+Generally speaking, an important keyword tend to be included in blog title, so if ``title`` contains ``Groonga``, its ``_score`` must be raised.
```
Comment (komainu8) [inline doc/source/tutorial/match_columns.rst]: In my opinion, we should clearly describe needing an index for using this feature.

File: doc/source/tutorial/match_columns.rst [hunk 2]
```diff
@@ -290,6 +290,18 @@ On the other hand, the second one matches one article only because of ``Replies2
...
 
-.. TODO: match_columnsの重み指定機能についても触れる。
+Here is the example which search blog entries with ``Groonga`` as important keyword in ``title`` or ``message`` columns.
+
+.. groonga-command
+.. include:: ../example/tutorial/match_columns-indexes-with-weight.log
+.. select --table Blog1 --match_columns 'IndexBlog1.index_title * 10 || IndexBlog1.index_message * 1' --query 'Groonga' --output_columns "_id, _score, *"
```
Comment (komainu8) [inline doc/source/tutorial/match_columns.rst]: In my opinion, we should add an example of the execution of the setting of the index.

File: doc/source/tutorial/match_columns.rst [hunk 3]
```diff
@@ -290,6 +290,18 @@ On the other hand, the second one matches one article only because of ``Replies2
...
+
+In above query, ``'IndexBlog1.index_title * 10 || IndexBlog1.index_message * 1'`` is specified for ``--match_columns``.
```
Comment (komainu8) [inline doc/source/tutorial/match_columns.rst]: Why do we set a default value of weight?
If we set it or not, I think the result of this query does not change.

File: doc/source/tutorial/match_columns.rst (hunk 1 above)
Comment (kenhys) [reply doc/source/tutorial/match_columns.rst]: I got it.

File: doc/source/tutorial/match_columns.rst (hunk 2 above)
Comment (kenhys) [reply doc/source/tutorial/match_columns.rst]: I've added reference.

File: doc/source/tutorial/match_columns.rst (hunk 3 above)
Comment (kenhys) [reply doc/source/tutorial/match_columns.rst]: Sure, it may be redundant. I've removed it.

**Comment by kenhys** [comment]
@komainu8 I've rebased and fixed based on your feedback.
Could you review it again, please?

**Comment by kenhys** [comment]
@komainu8  Thanks!

**Pull request merged**